### ISSUE TITLE
Remove while loop from differential evolution's r_draw

### ISSUE THREAD

**Issue by MichaelClerx** [pints-team/pints#1251]
I think this
```
    r1, r2 = np.random.choice(n, 2, replace=False)
    while(r1 == i or r2 == i or r1 == r2):
        r1, r2 = np.random.choice(n, 2, replace=False)
    return r1, r2
```

would be better as
```
    options = np.arange(n - 1)
    options[i:] += 1
    return np.random.choice(options, 2, replace=False)
```
?

- test and benchmark
- replace
- search for other occurrences of r_draw

**Comment by pratik-99** [comment]
can you assign me this task and elaborate how can I solve this issue I am pretty new over here

**Comment by prateekpandey1234** [comment]
please assign me this issue i have done this one

**Comment by MichaelClerx** [comment]
Hello! Thanks for wanting to contribute. Please note there is more to this issue than just replacing the code above (so also test that both snippets do the same, benchmark to see if the second is actually faster, and search for other occurrences of r_draw). To get the code merged please fork, create a PR, make sure that all tests pass, and wait for someone from the pints team to review it.

**Comment by ben18785** [comment]
Closed by #1285
